pageserver/src/tenant/storage_layer/inmemory_layer/vectored_dio_read.rs

```diff
@@ -0,0 +1,916 @@
+use std::{
+    collections::BTreeMap,
+    sync::{Arc, RwLock},
+};
+
+use itertools::Itertools;
+use tokio_epoll_uring::{BoundedBuf, IoBufMut, Slice};
+
+use crate::context::RequestContext;
+
+mod sealed {
+    pub trait Sealed {}
+}
```

Comment:
What does this do?

Reply:
https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/

Reply:
TIL

Reply:
then again, not really needed